### PR TITLE
Warn if CRS is NA, don't error

### PR DESCRIPTION
**Pull request by mikemahoney218**
Simulated data will often have an NA CRS, as there is not a real transformation to map the simulated data onto the real world. Right now, that means spatialsample can't easily be used with simulated data without bizarre workarounds. This PR changes that to enable using spatialsample with simulated data.

The original intent of the error was two-fold: NA CRS don't cooperate with unit conversions (because it's not clear what units to convert values _into_), and non-sf objects will have an NA CRS and therefore could _mostly_ not be used by these functions. That said, it _was_ possible to use tibbles and similar with `spatial_buffer_vfold_cv()` before, if a user manually set `radius == NULL && buffer == NULL` (not missing/default arguments, but fully specified by the user). That was a weird loophole that we already heavily discouraged -- all our documentation (and the error message) says that users should use `vfold_cv()` instead of `spatial_buffer_vfold_cv()` if they don't want buffering. As such, requiring sf objects is technically a breaking change, but I would be surprised if anyone was doing it.

### CODE REVIEW

File: R/buffer.R [hunk 1]
```diff
@@ -38,6 +49,8 @@ buffer_indices <- function(data, indices, radius, buffer, call = rlang::caller_e
   # only run radius checks if radius is not NULL (to prevent NAs from >)
   run_radius <- !is.null(radius)
   if (run_radius && units::set_units(radius, NULL) > 0) {
+    # In case `buffer` has no units, assume it's in the same units as `data`
+    if (!identical(sf::st_crs(data), sf::NA_crs_)) units(radius) <- units(distmat)
     # In case `radius` has no units, assume it's in the same units as `data`
     units(radius) <- units(distmat)
```
Comment (hfrick) [inline R/buffer.R:55]: this looks like it's not quite what you intended?

File: R/spatial_vfold_cv.R [hunk 2]
```diff
@@ -82,7 +91,6 @@ spatial_buffer_vfold_cv <- function(data,
         i = "Use `NULL` for resampling without one of `radius` or `buffer`, like `radius = NULL, buffer = 5000`",
```
Comment (hfrick) [inline R/spatial_vfold_cv.R:91]:
```suggestion
        i = "Use `NULL` for resampling without one of `radius` or `buffer`, like `radius = NULL, buffer = 5000`.",
```

File: R/buffer.R (hunk 1 above)
Comment (mikemahoney218) [reply R/buffer.R:55]: Gah -- messed up the copy and paste, thanks

**Comment by mikemahoney218** [comment]
Addressed both the issue with `radius` and the error message punctuation. 2ccd06bceef32e021f3e45e2ec5a203f48d35ccb is probably overkill, but I realized I could simplify the code somewhat by factoring all the repeated checks we do into their own functions.

**Review by hfrick — APPROVED** [review]
Looks good! I think the check functions are a nice addition 👍

**Review by juliasilge — APPROVED** [review]
I agree with @hfrick -- nice addition!

**Pull request merged**